src/picosvg/svg_types.py

```diff
@@ -70,24 +70,33 @@ def _copy_common_fields(
         self.opacity = opacity
         self.transform = transform
 
-    def visible(self) -> bool:
+    def might_paint(self) -> bool:
+        """False if we're sure this shape will not paint. True if it *might* paint."""
         def _visible(fill, opacity):
             return fill != "none" and opacity != 0
             # we're ignoring fill-opacity
 
-        return _visible(self.fill, self.opacity) or _visible(
-            self.stroke, self.stroke_opacity
-        )
+        if _visible(self.stroke, self.stroke_opacity):
+            return True
```

Comment:
a stroke that only has a move also doesn't paint, no?

Reply:
what kind of person would make a stroke with only a move :D

Reply:
Shuffled; now we return false if all cmds are move